lmfit/printfuncs.py

```diff
@@ -132,6 +135,15 @@ def fit_report(inpars, modelpars=None, show_correl=True, min_correl=0.1,
         add("    reduced chi-square = %s" % getfloat_attr(result, 'redchi'))
         add("    Akaike info crit   = %s" % getfloat_attr(result, 'aic'))
         add("    Bayesian info crit = %s" % getfloat_attr(result, 'bic'))
+        if not result.errorbars:
+            add("### Warning: uncertainties could not be estimated: ###")
```

Comment:
@newville this is a useful addition, perhaps we can add the few lines below (or similar) as well? That would inform the user uncertainties are only natively supported by ```leastsq``` and ```least_squares``` and that they should install ```numdifftools``` if they would like to have these estimated when using another minimization methods.
```
if not (result.method in ['leastsq', 'least_squares'] and HAS_NUMDIFFTOOLS):
    add("The minimization method does not calculate uncertainties.")
    add("Please install the numdifftools package if you want lmfit")
    add("estimate the uncertainties.")```

Reply:
@reneeotten Good point, I will add that.  Should we just require numdifftools?  I believe it is pure python, so pretty easy to pip install...

Reply:
I don't know... I have no strong preference either way. There are other packages (e.g., emcee, matplotlib, dill) that are useful, but not strictly necessary to use lmfit - we don't require them either. I think numdifftools falls in this category as well and, if anything, we should re-evaluate all of those. If, for example, making them requirements would allow us to simplify code then that would be something to consider.